library/core/class.request.php

```diff
@@ -86,6 +86,22 @@ class Gdn_Request implements RequestInterface {
     /** @var array Request data/parameters, either from superglobals or from a custom array of key/value pairs. */
     protected $_RequestArguments;
 
+    /**
+    /*  Holds domain plus port as a string to avoid multiple calculations
+     * https://github.com/vanilla/vanilla/issues/7617
+     *
+     * @var string ex: vanilla.local:8080
+     */
+    public static $host = false;
```

Comment:
I need to detect if $host was initiated, so I can't use '' (empty string) as default. And I don't like ```null``` as default because in many cases null means _does not exist_ || _not defined_. In this case ```false``` clearly alerting that property exists but not initialized yet. So not sure if we need to change anything here. About the name itself - what name do you prefer instead of **host**?

Reply:
To me, if you aren't defining a value...then you shouldn't define a value. You can declare this property without defining a default value. Your `empty` check would still work for this. A `null` value should indicate the absence of a "real" value. If you're dead set on setting a boolean `false` as your default, you should include boolean as a type for this parameter, since you are making booleans a valid value.

Reply:
> what name do you prefer instead of **host**?

I don't have any particular preferences, but "host" is already established to mean "only the domain name" in this class. It explicitly does not include the port. This new "host" property would potentially include the port. I'm not picky on a new name, but I do think it needs something to differentiate it.

Reply:
Done